dbt_checkpoint/check_model_columns_have_meta_keys.py

```diff
@@ -0,0 +1,193 @@
+
+    # get manifest nodes that pre-commit found as changed
+    models = get_models(manifest, filenames, include_disabled=include_disabled)
+    # if user added schema but did not rerun the model
+    schemas = get_model_schemas(list(ymls.values()), filenames)
+    missing: Dict[str, Any] = {}
+    meta_set = set(meta_keys)
+
+    for item in itertools.chain(models, schemas):
+        missing_cols = {}  # pragma: no mutate
+        if isinstance(item, ModelSchema):
+            model_name = item.model_name
+            missing_cols = {
+                key.get("name"): {
+                    "missing_meta_keys": sorted(
+                        [
+                            meta_key
+                            for meta_key in meta_set
+                            if meta_key not in key.get("meta", {}).keys()
+                        ]
+                    ),
+                    "extra_meta_keys": sorted(
+                        [
+                            extra_meta_key
+                            for extra_meta_key in key.get("meta", {}).keys()
+                            if extra_meta_key not in meta_set
+                        ]
+                    ),
+                }
+                for key in item.schema.get("columns", [])
+                if not validate_meta_keys(
+                    key.get("meta", {}).keys(), meta_set, allow_extra_keys
+                )
+            }
+        # Model
+        elif isinstance(item, Model):
+            model_name = item.filename
+            missing_cols = {
+                key: {
+                    "missing_meta_keys": sorted(
+                        [
+                            meta_key
+                            for meta_key in meta_set
+                            if meta_key not in value.get("meta", {}).keys()
+                        ]
+                    ),
+                    "extra_meta_keys": sorted(
+                        [
+                            extra_meta_key
+                            for extra_meta_key in value.get("meta", {}).keys()
+                            if extra_meta_key not in meta_set
+                        ]
+                    ),
+                }
+                for key, value in item.node.get("columns", {}).items()
+                if (
+                    not value.get("meta")
+                    or not validate_meta_keys(
+                        value.get("meta", {}).keys(), meta_set, allow_extra_keys
+                    )
+                    if value.get("meta")
+                    else True
+                )
+            }
+        else:
+            continue
+        seen = missing.get(model_name)
+        if seen:
+            if not missing_cols:
+                missing[model_name] = {}  # pragma: no mutate
+            else:
+                missing[model_name] = seen.union(missing_cols)
+        elif missing_cols:
+            missing[model_name] = missing_cols
+    for model, columns in missing.items():
```

Comment:
Many details from L126 to L142:
- Couldn't results be calculated above, while traversing Model and ModelSchemas?
- Result string can surely be better handled (a multi-line template, f-strings, some auxiliary in `utils.py`, etc.)
- Aren't we open to many KeyErrors? Take into account that dict's `get` method is almost always a better approach than `if key in dict` followed by direct `[key]` access.